craft_parts/executor/step_handler.py

```diff
@@ -158,14 +158,14 @@ def _builtin_stage(self) -> StepContents:
         srcdir = str(self._part.part_install_dir)
         files, dirs = filesets.migratable_filesets(stage_fileset, srcdir)
 
-        def pkgconfig_fixup(file_path):
+        def pkgconfig_fixup(file_path: str) -> None:
```

Comment:
```suggestion
        def pkgconfig_fixup(file_path: Union[os.PathLike, str]) -> None:
```

Reply:
but then it'd fail on line 164:
```python
            if not file_path.endswith(".pc"):
                return
```

Reply:
I read the function twice and still managed to miss that. Sorry!